### PR TITLE
Add Onboarding Documentation

### LINKED ISSUE #61: Better Developer Onboarding
When onboarding a new contributor, they had some questions which could be answered by good documentation. I'd like to make as easy as possible for people to contribute and understand what's going on.

## Generated Files
IDEs complain about generated files not being present. Builds will fail unless they are generated. The following should be documented:

* What generated files are and when they need to be updated
* Where they are

## Mock Data

Mock Data is pulled in from "@forte-music/mock". The following should be documented:

* Where does the mock data come from? (GitHub page and npm package)

## Audio Track

There's a single audio track which all songs point to. This is so playback works with mock data. This file is stored in Git LFS. Git LFS is overkill for this one file. We should find somewhere else to source the files from and have more than one (#44). For now we should update the documentation:

* How to setup git lfs and pull down this file?

## Real Data Sourcing

* Where does the audio come from when not in development?

## Sharing

* How is stuff shared between devices without this thing being on the internet?

## Developer Tooling

Explain what tools we use:

* `create-react-app`
* `storybook`
* `typescript`
* `prettier`
* `redux`
* `apollo`
* `graphql`
* `styled`

and how we lay out our project.

* render props pattern (https://reactjs.org/docs/render-props.html)
* everything is a component
* folder structure
* render props
* redux wrapper
* add example for redux wrapper

### PR DESCRIPTION
**Pull request by 0xcaff**
Waiting for #67

Closes #61

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Thanks for taking the time to contribute to forte. Here are some guidelines to follow when contributing.
+
+Some of the code in this project hasn't been refactored to follow these guidelines yet.
+
+## CSS
+
+For the most part, CSS is written using [styled-components]. Currently, they are scattered near the presentational components they are used in, in `src/components/styled` and in `src/components`. Going forward, mixins should be placed in `src/styled-mixins`, shared styled components in `src/components/styled` and unshared styled components, near where they are used.
+
+Avoid using theming unless needed. They add un-needed complexity for most use cases. If you are using the same component with slightly different styles in two different places, then use themeing.
```
Comment (Mcat12) [inline CONTRIBUTING.md]: "theming" is singular, so change "They add" to "It adds".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Redux
+The [render-props] pattern is used to decouple data providing components with components consuming data. We use a wrapper around `connect` (a Higher Order Component from `react-redux`) to use the render props pattern with redux. This wrapper is called `createReduxComponent` and can be found in `src/redux/render.tsx`.
+
+## Component Folder Structure
+There are a few different types of components.
+
+* Presentational Components
+  Stateless components which are primarily concerned with how things look. State and actions are passed as props and prop callbacks respectively. These are usually functional components. Usually placed in `src/components/`.
```
Comment (Mcat12) [inline CONTRIBUTING.md]: Markdown combines these two lines, so add a colon or some other separator after the first line.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Code Style
+
+[Prettier] is used to keep our code formatted consistently. Run `yarn fix-style` to reformat code to follow adhere to Prettier's style. Run `yarn check-style` to see which files don't adhere to Prettier's style.
+
+## Storybook
+
+[Storybook] is used for testing react components outside of the complete application. Run `yarn storybook` to start it with mock data. Files ending with `.stories.tsx` are picked up storybook. A test ensures that all stories render without crashing.
+
+## Tests
```
Comment (Mcat12) [inline CONTRIBUTING.md]: You have a "Linting" header, so this should be called "Testing", unless you want to change "Linting" to "Lints"

**Comment by 0xcaff** [comment]
Just a heads up, I'd like to get @chris-chao 's feedback before merging this.

**Comment by chris-chao** [comment]
The changes look good to me.

**Pull request merged**